services/skipchain/skipchain.go

```diff
@@ -0,0 +1,321 @@
+package skipchain
+
+import (
+	"crypto/rand"
+	"errors"
+
+	"fmt"
+
+	"bytes"
+
+	"sync"
+
+	"github.com/dedis/cothority/lib/cosi"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+// ServiceName can be used to refer to the name of this service
+const ServiceName = "Skipchain"
+
+func init() {
+	sda.RegisterNewService(ServiceName, newSkipchainService)
+	skipchainSID = sda.ServiceFactory.ServiceID(ServiceName)
+}
+
+var skipchainSID sda.ServiceID
+
+// Service handles adding new SkipBlocks
+type Service struct {
+	*sda.ServiceProcessor
+	// SkipBlocks points from SkipBlockID to SkipBlock but SkipBlockID is not a valid
+	// key-type for maps, so we need to cast it to string
+	SkipBlocks map[string]*SkipBlock
```

Comment:
I think Golang does not allow this is because a slice is not comparable [0]. One solution to this is to use a fixed size array, and we can do that since SkipBlockID is only a hash, and a hash has a fixed size output (32 with sha256).
Having 
```
type HashID [32]byte
type SkipBlockID HashID
```
can solve it I guess.

[0] : https://blog.golang.org/go-maps-in-action